destroy openstack server group operation

derekolk opened the pull request:
Implements issue spinnaker/spinnaker#928

emjburns commented on clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/converters/servergroup/DeployOpenstackAtomicOperationConverter.groovy:
```diff
@@ -5,12 +5,12 @@
  * You may obtain a copy of the License at
  *
- *    http://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
Why the formatting change here?

derekolk replied:
@emjburns The project level editor config is changing this.  I can switch back if needed but think it is right.

emjburns commented on clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/converters/servergroup/DestroyOpenstackAtomicOperationConverter.groovy:
```diff
@@ -0,0 +1,40 @@
+import org.springframework.stereotype.Component
+
+@OpenstackOperation(AtomicOperations.DESTROY_SERVER_GROUP)
+@Component("destroyOpenstackDescription")
```
We can probably just do @Component, no need to specify name

derekolk replied:
Will remove.

emjburns commented on clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DestroyOpenstackAtomicOperation.groovy:
```diff
@@ -0,0 +1,68 @@
+
+@Slf4j
+class DestroyOpenstackAtomicOperation implements AtomicOperation<Void> {
+  private final String BASE_PHASE = "DEPLOY"
```
Should this base_phase be "DESTROY"?

derekolk replied:
Good catch will change.

varikin commented on clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy:
```diff
@@ -177,13 +157,34 @@ abstract class OpenstackClientProvider {
+    Stack stack = handleRequest {
+      client.useRegion(region).heat().stacks().getStackByName(stackName)
+    }
+    if (stack == null) {
```
`!stack`

varikin commented on clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/OpenstackServerGroupNameResolver.groovy:
```diff
@@ -62,4 +62,10 @@ class OpenstackServerGroupNameResolver extends AbstractServerGroupNameResolver {
+
+  @Override
+  String combineAppStackDetail(String appName, String stack, String detail) {
+    super.combineAppStackDetail(appName, stack, detail)
```
Why do you have to override this just to call the super method?

varikin commented on clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/converters/servergroup/DeployOpenstackAtomicOperationConverter.groovy:
```diff
@@ -5,12 +5,12 @@
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
+ *  See the License for the specific language governing permissions and
```
I wonder if your license set in Intellij slightly differs from ours in formatting.

varikin commented on clouddriver-openstack/src/test/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/validators/servergroup/DestroyOpenstackAtomicOperationValidatorSpec.groovy:
```diff
@@ -0,0 +1,119 @@
+
+/**
+ *
+ */
```
Can you remove this empty javadoc?

varikin commented on clouddriver-openstack/src/test/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DestroyOpenstackAtomicOperationSpec.groovy:
```diff
@@ -0,0 +1,87 @@
+
+/**
+ *
+ */
```
Empty Javadoc here too

derekolk commented:
@varikin @emjburns @lwander - Addressed code review comments of using groovy truthiness, removed license formatting changes, and removed any empty code comments.

lwander commented:
LGTM, I'll merge once squashed

derekolk commented:
@lwander squashed and ready to be merged.

lwander commented:
Could you pick a more descriptive commit title? That's what will be shown in the history, and "checkpoint" isn't ideal.

derekolk commented:
Changed the PR and commit to be more descriptive and refer to github issue.

lwander commented:
Needs a rebase, looks like someone snuck in a merge before us.

derekolk commented:
@lwander rebased.
